### PR TITLE
feat: Add new tokens to `SearchInput` [FS-496]

### PR DESCRIPTION
**Pull request by lucasfp13**
## What's the purpose of this pull request?

This PR intends to apply new tokens to `SearchInput` based on the new [Theming structure](https://github.com/vtex-sites/base.store/pull/407): new scoped tokens linked to global ones.

## How does it work?

This PR uses local variables to parameterize the `SearchInput` properties, then connects these scoped tokens to the global ones.

## How to test it?

These changes should not affect the `SearchInput` behavior. It can be tested through Storybook, inside the `Search` feature page.

## Checklist

**PR Title and Commit Messages**
- [x] PR title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification
- [x] Commit messages follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*.

### CODE REVIEW

File: src/components/search/SearchInput/search-input.module.scss [hunk 1]
```diff
@@ -0,0 +1,154 @@
+@import "src/styles/scaffold";
+
+.fs-search-input {
+  // --------------------------------------------------------
+  // Design Tokens for Search Input
+  // --------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/search/SearchInput/search-input.module.scss:6]: Agreed!! Current, there is no tokens listed [there](https://nextjs-store-storybook-git-feat-fs-496-add-new-7f25aa-faststore.vercel.app/?path=/docs/features-search-searchinput--default-story).

File: src/components/search/SearchInput/SearchInput.tsx [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-import type { CSSProperties } from 'react'
 import {
-  forwardRef,
   lazy,
-  Suspense,
   useRef,
   useState,
+  forwardRef,
   useDeferredValue,
+  Suspense,
```
Comment (hellofanny) [inline src/components/search/SearchInput/SearchInput.tsx:7]: Is this being reorder automatically?

File: src/components/search/SearchInput/search-input.module.scss [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+  // Default properties
+  --fs-search-input-width                         : 100%;
```
Comment (hellofanny) [inline src/components/search/SearchInput/search-input.module.scss:9]: probably, we won't need this token 🤔

File: src/components/search/SearchInput/SearchInput.tsx (hunk 2 above)
Comment (lucasfp13) [reply src/components/search/SearchInput/SearchInput.tsx:7]: No, it was me just reordering the imports to leave related imports close to each other

File: src/components/search/SearchInput/search-input.module.scss [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  --fs-search-input-height-mobile                 : var(--fs-control-tap-size);
+  --fs-search-input-height-desktop                : var(--fs-spacing-6);
+
+  --fs-search-input-border-width                  : var(--fs-border-width);
+  --fs-search-input-border-radius                 : var(--fs-border-radius);
+  --fs-search-input-border-color                  : var(--fs-border-color);
+  --fs-search-input-border-color-hover            : var(--fs-border-color-active);
+
+  --fs-search-input-timing-timing                 : var(--fs-transition-timing);
+  --fs-search-input-timing-function               : ease;
```
Comment (hellofanny) [inline src/components/search/SearchInput/search-input.module.scss:22]:
```suggestion
  --fs-search-input-transition-timing                 : var(--fs-transition-timing);
  --fs-search-input-transition-function               : ease;
```

File: src/components/search/SearchInput/SearchInput.stories.mdx [hunk 5]
```diff
@@ -55,3 +55,151 @@ Responsible to receive, search and display suggestions.
...
+    value="var(--fs-border-color)"
+  />
+  <TokenRow
+    isColor
+    token="--fs-search-input-border-color-hover"
+    value="var(--fs-border-color-active)"
+  />
+  <TokenDivider />
+  <TokenRow
+    token="--fs-search-input-timing-timing"
```
Comment (hellofanny) [inline src/components/search/SearchInput/SearchInput.stories.mdx:103]:
```suggestion
    token="--fs-search-input-transition-timing"
```

File: src/components/search/SearchInput/SearchInput.stories.mdx [hunk 6]
```diff
@@ -55,3 +55,151 @@ Responsible to receive, search and display suggestions.
...
+    value="var(--fs-transition-timing)"
+  />
+  <TokenRow token="--fs-search-input-timing-function" value="ease" />
```
Comment (hellofanny) [inline src/components/search/SearchInput/SearchInput.stories.mdx:106]:
```suggestion
  <TokenRow token="--fs-search-input-transition-function" value="ease" />
```

File: src/components/search/SearchInput/search-input.module.scss [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+    @include media(">=notebook") { height: var(--fs-search-input-height-desktop); }
+
+    [data-fs-search-input] {
+      width: 100%;
+      padding: var(--fs-search-input-padding);
+      background-color: var(--fs-search-input-bkg-color);
+      border: var(--fs-search-input-border-width) solid var(--fs-search-input-border-color);
+      border-radius: var(--fs-search-input-border-radius);
+      transition: box-shadow var(--fs-search-input-timing-timing) var(--fs-search-input-timing-timing), border var(--fs-search-input-timing-timing) var(--fs-search-input-timing-function);
```
Comment (hellofanny) [inline src/components/search/SearchInput/search-input.module.scss:75]:
```suggestion
      transition: box-shadow var(--fs-search-input-transition-timing) var(--fs-search-input-transition-timing), border var(--fs-search-input-transition-timing) var(--fs-search-input-transition-function);
```

**Review by hellofanny — APPROVED** [review]
🌟

**Pull request merged**